### DISCUSSION TITLE
Recommended Features

### DISCUSSION THREAD

**Discussion by ProfSynapse** [brianpetro/obsidian-smart-connections#408] [Ideas]
Brian, you've really created something special here. This is by far the most game changing application let alone community plugin. 

I have a couple feature requests to really send this over the moon.

1. Add the ability for the user to create the system prompt: I do manually go into main and change it manually, but then every time there's an update it reverts back obviously, and this is also harder for people who don't have much coding knowledge. This is essential for me as I do some things like have it always reference a specific note before outputting. Similarly being bale to change the welcome message would be nice.
2. Add the ability to turn the knobs: It would be great to be able to manually control the temp, top_p, output length, etc within the chat itself even. 

Do these two things, and I'll give you all my money!

**Comment by brianpetro** [comment]
Hi @ProfSynapse 

Thanks for your kind words and suggestions! Here's a bit of a breakdown of the current roadmap:

> Add the ability for the user to create the system prompt

This will be available in the upcoming `v2.1` when I rewrite the Smart Chat feature.

> Similarly being bale to change the welcome message would be nice.

I haven't considered this one. I'll see what I can do!

> Add the ability to turn the knobs

A token-length "knob" will definitely make it into `v2.1`. Likely a knob for each input and output token length.

A temperature knob is likely and would be implemented based on a preset low/medium/high that's configured for each available model.

Other parameters are less likely to make it into the initial `v2.1` release.

Let me know what you think.
🌴

**Reply by neomagic75** [reply]
An easy way to pick different notes as a system message would turn the whole thing into a custom GPT hub👍
